### PR TITLE
added sql db transparent data encryption enabled

### PR DESCRIPTION
**Pull request by muzzamilinovaqo**

### CODE REVIEW

File: plugins/azure/sqldatabases/dbTDEEnabled.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Transparent Data Encryption Enabled',
+    category: 'SQL Databases',
+    domain: 'Databases',
+    description: 'Ensure Transparent Data Encryption (TDE) is enabled on SQL databases for enhanced security',
+    more_info: 'TDE helps protect sensitive data at rest by encrypting the database files.',
```
Comment (fatima99s) [inline plugins/azure/sqldatabases/dbTDEEnabled.js:9]: can we add more infomation or advantages of Transparent Data Encryption

File: exports.js [hunk 2]
```diff
@@ -989,7 +990,7 @@ module.exports = {
         'eventHubMinimumTLSversion'     : require(__dirname + '/plugins/azure/eventhub/eventHubMinimumTLSversion.js'),
 
         'accessLogsEnabled'             : require(__dirname + '/plugins/azure/frontdoor/accessLogsEnabled.js'),
-        'frontDoorMinimumTlsVersion'    : require(__dirname + '/plugins/azure/frontdoor/frontDoorMinimumTlsVersion.js')
+        'frontDoorMinimumTlsVersion'    : require(__dirname + '/plugins/azure/frontdoor/frontDoorMinimumTlsVersion.js'),
```
Comment (alphadev4) [inline exports.js:993]:
```suggestion
        'frontDoorMinimumTlsVersion'    : require(__dirname + '/plugins/azure/frontdoor/frontDoorMinimumTlsVersion.js')
```

File: plugins/azure/sqldatabases/dbTDEEnabled.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Transparent Data Encryption Enabled',
+    category: 'SQL Databases',
+    domain: 'Databases',
+    description: 'Ensure Transparent Data Encryption (TDE) is enabled on SQL databases.',
```
Comment (alphadev4) [inline plugins/azure/sqldatabases/dbTDEEnabled.js:8]:
```suggestion
    description: 'Ensure that Transparent Data Encryption (TDE) is enabled for SQL databases.',
```

File: plugins/azure/sqldatabases/dbTDEEnabled.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    more_info: 'Transparent data encryption (TDE) helps protect Azure SQL Database, Azure SQL Managed Instance, and Azure Synapse Analytics against the threat of malicious offline activity by encrypting data at rest. It performs real-time encryption and decryption of the database, associated backups, and transaction log files at rest without requiring changes to the application.',
```
Comment (alphadev4) [inline plugins/azure/sqldatabases/dbTDEEnabled.js:9]:
```suggestion
    more_info: 'Transparent data encryption (TDE) helps protect SQL Database, SQL Managed Instance, and Azure Synapse Analytics against the threat of malicious offline activity by encrypting data at rest. It performs real-time encryption and decryption of the database, associated backups, and transaction log files at rest without requiring changes to the application.',
```

File: plugins/azure/sqldatabases/dbTDEEnabled.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+                helpers.addResult(results, 3, 'Unable to query for SQL servers: ' + helpers.addError(servers), location);
+                return rcb();
+            }
+
+            if (!servers.data.length) {
+                helpers.addResult(results, 0, 'No SQL servers found', location);
+                return rcb();
+            }
+
+            servers.data.forEach(server=> {
```
Comment (alphadev4) [inline plugins/azure/sqldatabases/dbTDEEnabled.js:34]:
```suggestion
            servers.data.forEach(server => {
```

File: plugins/azure/sqldatabases/dbTDEEnabled.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+                if (!databases || databases.err || !databases.data) {
+                    helpers.addResult(results, 3,
+                        'Unable to query for SQL server databases: ' + helpers.addError(databases), location, server.id);
+                } else {
+                    if (!databases.data.length) {
+                        helpers.addResult(results, 0,
+                            'No databases found for SQL server', location, server.id);
+                    } else {
+                        databases.data.forEach(database=> {
```
Comment (alphadev4) [inline plugins/azure/sqldatabases/dbTDEEnabled.js:46]:
```suggestion
                        databases.data.forEach(database => {
```

File: plugins/azure/sqldatabases/dbTDEEnabled.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+                            if (!transparentDataEncryption || transparentDataEncryption.err || !transparentDataEncryption.data || !transparentDataEncryption.data.length) {
+                                helpers.addResult(results, 3, 'Unable to query transparent data encryption for SQL Database: ' + helpers.addError(transparentDataEncryption), location, database.id);
+                                return;
+                            }
+
+                            if (transparentDataEncryption.data[0].state.toLowerCase()=='enabled') {
+                                helpers.addResult(results, 0, 'Transparent data encryption is enabled for SQL Database', location, database.id);
+                            } else {
+                                helpers.addResult(results, 2, 'Transparent data encryption is not enabled for SQL Database', location, database.id);
+                            }
```
Comment (alphadev4) [inline plugins/azure/sqldatabases/dbTDEEnabled.js:59]:
```suggestion
                            var encryption = transparentDataEncryption.data[0];
                            if (encryption.state && encryption.state.toLowerCase() == 'enabled') {
                                helpers.addResult(results, 0, 'Transparent data encryption is enabled for SQL Database', location, database.id);
                            } else {
                                helpers.addResult(results, 2, 'Transparent data encryption is not enabled for SQL Database', location, database.id);
                            }
```

**Review by alphadev4 — APPROVED** [review]

**Review by mehakseedat63 — APPROVED** [review]

**Pull request merged**